staging/src/k8s.io/apiserver/pkg/util/dryrun/dryrun.go

```diff
@@ -16,7 +16,32 @@ limitations under the License.
 
 package dryrun
 
+import (
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
 // IsDryRun returns true if the DryRun flag is an actual dry-run.
 func IsDryRun(flag []string) bool {
 	return len(flag) > 0
 }
+
+func RevertDryRunMetadata(originalObj, newObj runtime.Object) error {
+	originalObjMeta, err := meta.Accessor(originalObj)
+	if err != nil {
+		return errors.NewInternalError(err)
+	}
+	newObjMeta, err := meta.Accessor(newObj)
+	if err != nil {
+		return errors.NewInternalError(err)
+	}
+	if originalObjMeta.GetGenerateName() != "" {
+		newObjMeta.SetName("")
+	}
+	newObjMeta.SetGenerateName(originalObjMeta.GetGenerateName())
+	newObjMeta.SetUID("")
```

Comment:
```suggestion
	// If the UID is a non-empty string, then non-dry-run-creation will fail. To allow non-dry-run creation of the object returned by dry-run creation, we set the UID to an empty string.
	newObjMeta.SetUID("")
```

Reply:
Reworded as I was getting confused switching back and forth between dry-run and non-dry-run. WDYT?